app/Models/Signup.php

```diff
@@ -213,17 +213,29 @@ public function getAcceptedQuantity()
     }
 
     /**
-     * Scope a query to only return signups if a user is an admin, staff, or is owner of signup.
+     * Scope a query to only return signups if a user is an admin, staff, or is owner of signup and by type (optional)
      *
+     * @param array $types
      * @return \Illuminate\Database\Eloquent\Builder
      */
-    public function scopeWithVisiblePosts($query)
+    public function scopeWithVisiblePosts($query, $types = null)
     {
-        if (! is_staff_user()) {
-            return $query->with(['posts' => function ($query) {
-                $query->where('status', 'accepted')
-                ->orWhere('northstar_id', auth()->id());
-            }]);
-        }
+        return $query->with(['visiblePosts' => function ($query) use ($types) {
+
+            if ($types) {
+                // $query->whereIn('type', $types);
+
+                if (count($types) >1 ) {
+                    $query->where(function ($query) use ($types) {
+                        foreach ($types as $type) {
+                            $query->orWhere('type', $type);
+                        }
+                    });
+                } else {
+                    $query->where('type', $types[0]);
+                }
+            }
+
```

Comment:
@DFurnes also tried this way too but still not returning posts with the matching `types` I'm sending in. I checked clockwork and it does look like the queries are potentially overriding the `type` match (if the order goes from oldest query at the top and latest query at the bottom of the list)?

Reply:
The queries look fine to me at first glance. First, we get the matching signups & then we make one more query for each of the relationships we're included (posts, reactions, and tags):

![screen shot 2018-07-23 at 4 43 18 pm](https://user-images.githubusercontent.com/583202/43101984-d6904f52-8e97-11e8-87c6-8ea9cc6b6979.png)

Reply:
Our problem is somewhere in that first "get posts for each signup" query, where it's not properly scoping things to the provided posts.

Reply:
@DFurnes I don't _think_ this matters but is it because it isn't doing `posts.type` and just doing `type` in the query?

Reply:
actually I don't think so because `status` is just there alone 🤔

Reply:
I think our problem is this first `status = 'accepted' or northstar_id IS NULL` in here, which is coming from the `visiblePosts` relationship:

```sql
SELECT `posts`.*, (SELECT count(*) FROM `reactions` WHERE `posts`.`id` = `reactions`.`post_id` and `reactions`.`deleted_at` IS NULL) as `reactions_count` FROM `posts` WHERE (`status` = 'accepted' or `northstar_id` IS NULL and `posts`.`signup_id` in ('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21') and `status` = 'accepted' and `type` in ('voter-reg', 'text')) and `posts`.`deleted_at` IS NULL
```

When I update that relationship to check similarly to our eager load method in 6b5313a, things look like they start working (both in the failing test & manual testing with Paw).

That leaves me with just one failing test:

```
1) Tests\Http\SignupTest::testSignupIndexWithIncludedPostsAndParamsAsAdmin
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'text'
+'photo'

/home/vagrant/Sites/rogue/tests/Http/SignupTest.php:381
```

Reply:
ah interesting, @DFurnes ! Thanks for this!! So does this have to re-check in the `scopeWithVisiblePosts` after it gets the results from `visiblePosts`? 

Re: the failing test, that is a funny one too! When I run that by itself, it seems to pass but only failed with running all tests together. In the wercker build it seemed to pass ok?

Reply:
After running it a few more times, I think that test may just be flaky based on timing (since we're checking specifically the first post is type `text` and the second one is type `photo`). Instead, let's check the number of responses we get with different filters (updated in b82192a6).